trie/proof.go

```diff
@@ -39,7 +41,11 @@ func (t *Trie) Prove(key []byte) []rlp.RawValue {
 		case nil:
 			return nil
 		case hashNode:
-			tn = t.resolveHash(n)
+			var err error
+			tn, err = t.resolveHash(n, nil, nil)
+			if err != nil && glog.V(logger.Error) {
+				glog.Errorf("Unhandled trie error: %v", err)
```

Comment:
Maybe `return nil` here? The proof will not be useful if the trie is incomplete.